pkg/sqlcache/informer/listoption_indexer_test.go

```diff
@@ -2245,11 +2250,14 @@ func TestWatchResourceVersion(t *testing.T) {
 			watcherCh, errCh := startWatcher(ctx, loi, test.rv)
 			gotEvents := receiveEvents(watcherCh)
 
-			assert.Equal(t, test.expectedEvents, gotEvents)
-
 			cancel()
 			err := waitStopWatcher(errCh)
-			assert.NoError(t, err)
+			if test.expectedErr != nil {
+				assert.ErrorIs(t, err, ErrTooOld)
+			} else {
+				assert.Equal(t, test.expectedEvents, gotEvents)
```

Comment:
This block should be
```
require.NoError(t, err)
assert.Equal(t, test.expectedEvents, gotEvents)
```

Assuming if we aren't expecting an error and we get one, comparing the expected events first will give a confusing result.

Reply:
Good point, fixed